imap/ctl_mboxlist.c

```diff
@@ -882,6 +883,91 @@ static void do_verify(void)
     mboxlist_allmbox("", &verify_cb, &found, MBOXTREE_TOMBSTONES);
 }
 
+struct root_rock {
+    mbname_t *mbname;
+    char *mboxid;
+};
+
+static int addroots_cb(const mbentry_t *mbentry, void *rockp)
+{
+    mbname_t *mbname = mbname_from_intname(mbentry->name);
+    struct root_rock *rrock = rockp;
+    int is_root = 0, is_child = 0;
+
+    /* Skip DELETED.* mailboxes */
+    if (mbname_isdeleted(mbname)) {
```

Comment:
I think maybe we do want to add the INBOX root_mailboxid to DELETED.user.foo mailboxes as well, so we know to consider them in the same userid, since they're findable with mboxlist_usermboxtree and flags.